public/persist.php

```diff
@@ -136,6 +211,134 @@ function perform_request_remove($args, $indexJson) {
   return true;
 }
 
+/**
+ * Generate a random string of safe characters.
+ *
+ * @param int $length
+ * - The length for this string
+ * @return string
+ */
+function generate_random_string($length = 10) {
+    $characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+    $charactersLength = strlen($characters);
+    $randomString = '';
+    for ($i = 0; $i < $length; $i++) {
+        $randomString .= $characters[rand(0, $charactersLength - 1)];
+    }
+    return $randomString;
+}
+
+/**
+ * Generate a random string of safe characters.
+ *
+ * @param string $source
+ * @return string
+ */
+function obfuscate($source) {
+  $pre = substr($source, 0, -6);
+  $post = substr($source, -6);
+  return $pre . str_rot13($post);
+}
+
+/**
+ * Add this client to the "verify list".
+ *
+ * @param array $args
+ * - The arguments for this request.
+ * @return array
+ */
+function perform_request_new_client($args) {
+  $verifyFile = 'sites/verify.json';
+  $currentVerify = json_decode(file_get_contents($verifyFile));
+
+  foreach ($currentVerify as $current) {
+    if ($current->email == $args['email'] ||
+        $current->company == $args['company'] ||
+        $current->phone == $args['phone']) {
+      // Client has already requested access.
+      return false;
+    }
+  }
+
+  // Add this to the pending list.
+  $client = (object) $args;
+
+  $client->clientid = strtolower(preg_replace("/[^A-Za-z0-9 ]/", '', $client->company));
+  $client->key = $client->clientid . '-' . generate_random_string(6);
+  array_push($currentVerify, $client);
+  file_put_contents($verifyFile, json_encode($currentVerify));
+
+  // Send them an email about it.
+  $subject = 'Access to Website Crawler: ' . $client->company;
```

Comment:
User entered values should probably be sanitised `htmlentities($client->company)`